frontend/packages/knative-plugin/src/actions/providers.ts

```diff
@@ -91,6 +91,7 @@ export const useSinkPubSubActionProvider = (resource: K8sResourceKind) => {
 export const useKnativeServiceActionsProvider = (resource: K8sResourceKind) => {
   const [kindObj, inFlight] = useK8sModel(referenceFor(resource));
   const serviceTypeValue = React.useContext(KnativeServiceTypeContext);
+  const { podData } = usePodsWatcher(resource, referenceFor(resource), resource.metadata.namespace);
   const actions = React.useMemo(() => {
```

Comment:
`podData` value is always there and Test serverless function is not disabled. 

Instead of this solution, check the status of the service and if all the status conditions are not `True` then disable the button with a tooltip as "Service is not ready" or something like that.

https://github.com/user-attachments/assets/966c990c-5365-422f-8822-5c47734424f8